modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,106 @@
+// Module included in the following assemblies:
```

Comment:
We should add some more detail around backup and restore along with extra manifest, usecases etc. This doc may be a good starting point https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/image-based-upgrade.md